lib/index.js

```diff
@@ -300,3 +370,44 @@ Segment.prototype.referrerId = function(query, ctx) {
  */
 
 function noop() {}
+
+/**
+ * getCrossDomainId
+ */
+function getCrossDomainId(domain, writeKey, callback) {
+  var endpoint = 'https://' + domain + '/v1/id/';
+  getJsonWithAuth(endpoint, writeKey, '', function(err, res) {
+    callback({
+      domain: domain,
+      id: (res && res.id) || null
+    });
+  });
+}
+
+/**
+ * getJsonWithAuth
+ * TODO: Add some queuing (multiple requests) and retry?
+ */
+function getJsonWithAuth(url, username, password, callback) {
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', url, true);
+  if (username || password) {
+    xhr.setRequestHeader('Authorization', 'Basic ' + btoa(username + ':' + password));
```

Comment:
Btw we removed this as per https://github.com/segmentio/cross-domain-id-service#get-v1idwritekey